configure

```diff
@@ -248,6 +248,7 @@ my @specs = # Array of array-refs, each inner array is an option group which
     'object-model-profile!', 'Object Model Profile (yes)',
     'persistence-profile!', 'Persistence Profile (yes)',
     'safety-profile:s', 'Safety Profile: base or extended (none)',
+    'suppress-anys!', 'Suppress Anys (yes))',
```

Comment:
Looks like we need to decide how this should interact with the related MPC features.  I would expect it to have an impact if we're going to make it a configure option.

On the other hand, we want it to only impact how the application's IDL is built and not impact the core, then it shouldn't be in the configure script.  In that case it would be up to the build system (both CMake and MPC) of the client application to set the option (in CMake's case that could be a cache variable or an argument to a function or ??).

Reply:
configure is what sets up our config.cmake file which is where we had the OPENDDS_SUPPRESS_ANYS flag being set.  This allows the user to turn that flag off.  The other alternatives would be:

1.  The user manually editing the cmake.config file and turning OPENDDS_SUPPRESS_ANYS off after their configure has been run.

2. manually removing the flags in their cmake lists:
list(REMOVE_ITEM OPENDDS_DDS_BASE_IDL_FLAGS -Sa -St)
list(REMOVE_ITEM OPENDDS_TAO_BASE_IDL_FLAGS -Sa -St)

3. Updating our macros to have a macro to strip the -Sa -St flags like in 2. This macro would then need to be used in CMakeLists.txt.

Reply:
Let's go back to the question what exactly this option needs to do.  In my previous comment I listed two possible approaches.  I still think that either could work, so we need understand both approaches and their pros/cons.

1. If the user specifies `--no-suppress-anys` to the configure script, the effect of this would be to change how OpenDDS core libraries are built.  Not doing so would be really surprising since that's what the configure script is for.  To make this work, add `dds_suppress_anys=0` to the default.features before configure runs MPC.  However, this wide-reaching of a change may not be needed.  In which case we consider...

2. If the core libraries should not be changed, this kind of functionality could be provided by only changing how the application's IDL is built.  In this case, `configure` shouldn't change at all (which means `config.cmake` doesn't change either).  Instead, for each build system (MPC and CMake) provide (and test) a way for the user to specify that support for Anys and TypeCodes are needed only in these projects.

At this point I'd be more in favor of approach 2 since it minimizes impact (and footprint) of the change.  If it turns out the changing how the core libs are built is indeed necessary, that could be added in a subsequent PR.